### PR TITLE
run as non-root for operator.

### LINKED ISSUE #24815: [istio-operator] istioctl operator init not working
**Bug description**
I am following the operator install [instructions](https://istio.io/latest/docs/setup/install/standalone-operator/) on a clean kubernetes 1.17 cluster made with kubespray.

I ran the following commands:
```
➜ istioctl operator init
Using operator Deployment image: docker.io/istio/operator:1.6.3
- Processing resources for Istio operator. Waiting for Deployment/istio-operator/istio-operator
```

The `istio-operator` namespace was created, but the istio-operator pod was in error state
```
NAME                              READY   STATUS                       RESTARTS   AGE
istio-operator-8494bc7758-wr954   0/1     CreateContainerConfigError   0          16m
```
with the following events:

```
Events:
  Type     Reason     Age                  From                     Message
  ----     ------     ----                 ----                     -------
  Normal   Scheduled  <unknown>            default-scheduler        Successfully assigned istio-operator/istio-operator-8494bc7758-wr954 to node-2
  Normal   Pulling    17m                  kubelet, node-2  Pulling image "docker.io/istio/operator:1.6.3"
  Normal   Pulled     16m                  kubelet, node-2  Successfully pulled image "docker.io/istio/operator:1.6.3"
  Warning  Failed     14m (x12 over 16m)   kubelet, node-2  Error: container has runAsNonRoot and image will run as root
  Normal   Pulled     119s (x69 over 16m)  kubelet, node-2  Container image "docker.io/istio/operator:1.6.3" already present on machine
```

Any ideas why the operator container runs as root or, at least, tries to? 

Thank you!

[ ] Configuration Infrastructure
[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**

The operator to be installed successfully.

**Steps to reproduce the bug**

Described above.

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

```
➜ kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-13T18:08:14Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.6", GitCommit:"d32e40e20d167e103faf894261614c5b45c44198", GitTreeState:"clean", BuildDate:"2020-05-20T13:08:34Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```

```
➜ istioctl version --remote=false
1.6.3
```
**How was Istio installed?**

**Environment where bug was observed (cloud vendor, OS, etc)**
On-Premise

### PR DESCRIPTION
**Pull request by morvencao**
Please provide a description for what this PR is for.
resolve: https://github.com/istio/istio/issues/24815

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ X ] Configuration Infrastructure
[ ] Docs
[ X ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by sdake** [comment]
@howardjohn distroless does have nonroot, however, we build operator with base by default.

Cheers,
-steve

File: operator/docker/user_setup [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+set -x
+
+# ensure $HOME exists and is accessible by group 0 (we don't know what the runtime UID will be)
+echo "${USER_NAME}:x:${USER_UID}:0:${USER_NAME} user:${HOME}:/sbin/nologin" >> /etc/passwd
+mkdir -p "${HOME}"
+chown "${USER_UID}:0" "${HOME}"
+chmod ug+rwx "${HOME}"
+
+# no need for this script to remain in the image after running
+rm "$0"
```
Comment (sdake) [inline operator/docker/user_setup:11]: images (should now and will be) read-only in the future. I'd encourage no self-modification of the images in PRs.

File: operator/docker/user_setup [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+set -x
+
+# ensure $HOME exists and is accessible by group 0 (we don't know what the runtime UID will be)
+echo "${USER_NAME}:x:${USER_UID}:0:${USER_NAME} user:${HOME}:/sbin/nologin" >> /etc/passwd
```
Comment (sdake) [inline operator/docker/user_setup:5]: nologin isn't going to work for all cases. There is a (VERY) long debate about this in the distroless repo where `nonroot` was added.

File: operator/docker/Dockerfile.operator [hunk 3]
```diff
@@ -15,10 +15,18 @@ FROM gcr.io/distroless/static-debian10@sha256:4433370ec2b3b97b338674b4de5ffaef8c
 # hadolint ignore=DL3006
 FROM ${BASE_DISTRIBUTION}
 
+ENV USER_UID=1001 \
```
Comment (sdake) [inline operator/docker/Dockerfile.operator:18]: why not align with distroless (uid 65532) - as we won't easily be able to add UID/GID in a distroless future?

**Review by sdake — COMMENTED** [review]
I had hoped the assignment of the issue would ensure we didn't duplicate work - however- sadly work was duplicated :) If you want to take over the PR @morvencao - please assign yourself the issue.

Cheers,
-steve

**Comment by morvencao** [comment]
Closing this in favor of https://github.com/istio/istio/pull/24858

**Pull request closed without merging**